pkg/nfd-worker/nfd-worker.go

```diff
@@ -271,6 +314,33 @@ func (w *nfdWorker) Run() error {
 	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
 	defer labelTrigger.Stop()
 
+	// Create owner ref
+	ownerReference := []metav1.OwnerReference{}
+	// Get pod owner reference
+	podName := os.Getenv("POD_NAME")
+
+	selfPod, err := w.k8sClient.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		klog.InfoS("Pod not found, cannot inherit ownerReference for NodeFeature")
+	} else if err == nil {
+		ownerReference = append(ownerReference, selfPod.OwnerReferences...)
+	}
+
+	// Add pod owner reference if it exists
+	podUID := os.Getenv("POD_UID")
+	if podName != "" && podUID != "" {
+		ownerReference = append(ownerReference, metav1.OwnerReference{
+			APIVersion: "v1",
+			Kind:       "Pod",
+			Name:       podName,
+			UID:        types.UID(podUID),
+		})
+	} else {
+		klog.InfoS("Cannot append POD ownerReference to NodeFeature, POD_NAME and/or POD_UID not specified")
+	}
```

Comment:
nit/suggestion: I think this could be made a bit clearer (e.g. both owner refs depend on POD_NAME env). For example like:

```golang
	if podName != "" {
		if selfPod, err := w.k8sClient.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{}); err != nil {
			klog.ErrorS(err, "failed to get self pod, cannot inherit ownerReference for NodeFeature")
		} else {
			ownerReference = append(ownerReference, selfPod.OwnerReferences...)
		}

		podUID := os.Getenv("POD_UID")
		if podUID != "" {
			ownerReference = append(ownerReference, metav1.OwnerReference{
				APIVersion: "v1",
				Kind:       "Pod",
				Name:       podName,
				UID:        types.UID(podUID),
			})
		} else {
			klog.InfoS("Cannot append POD ownerReference to NodeFeature, POD_UID not specified")
		}
	} else {
		klog.InfoS("Cannot set NodeFeature owner references, POD_NAME not specified")
	}
```

WDYT?